apps/dashboard/src/@3rdweb-sdk/react/hooks/useDashboardOwnedNFTs.ts

```diff
@@ -0,0 +1,182 @@
+import { useQuery } from "@tanstack/react-query";
+import {
+  type BaseTransactionOptions,
+  type NFT,
+  type ThirdwebContract,
+  ZERO_ADDRESS,
+} from "thirdweb";
+import { isERC721 } from "thirdweb/extensions/erc721";
+import { getOwnedNFTs } from "thirdweb/extensions/erc1155";
+import { detectMethod } from "thirdweb/utils";
+import invariant from "tiny-invariant";
+
+/**
+ * This hook is created for 2 main purposes:
+ * 1. Replacing useOwnedNFTs from v4 so that we are unblocked for v5
+ * 2. Replacing indexer _on the chains that Indexer hasn't supported yet_,
+ * otherwise, an indexing solution should always be used for this purpose
+ *
+ * We can't use v5's ERC721 extension `getOwnedNFTs` because it only supports contract with
+ * the [`tokenOfOwnerByIndex`](https://portal.thirdweb.com/references/typescript/v5/erc721/tokenOfOwnerByIndex) method
+ */
```

Comment:
we should use it regardless and instead drop support for contracts that don't have it -> then we need to replace it with indexer solutions over time (likely directly in SDK) - re-implementing the change that breaks on large collections here instead of in SDK is not the path